Basic.wl

```diff
@@ -180,10 +161,36 @@ ImprisonmentExpressions[template_List]:= Cases[template, x_ \[Element] set_ ,Inf
 ValueRestrictions[imprisonmentExpression_]:=
  Apply[Or,imprisonmentExpression[[1]] == #&/@ imprisonmentExpression[[2]]];
 
+RemoveContainedSubsSets[allSubSet_List, subSet_List] :=
+ Module[{listToRemove, result},
+  listToRemove = ContainsAll[#, subSet] && subSet =!= # & /@ allSubSet;
+  result = MapThread[If[! #2, #1, ## &[]] &, {allSubSet, listToRemove}]
+];
+
+BooleanFromRule[x_ /; x[[2]] == 0] := Not[x[[1]]];
+BooleanFromRule[x_ /; x[[2]] == 1] := x[[1]];
+
+ConjunctionFromRuleSet[ruleSet_List] :=
+    Apply[And, Map[BooleanFromRule, ruleSet]];
+
+DNFFromRuleSet[ruleSet_List] :=
+    Apply[Or, Map[ConjunctionFromRuleSet, ruleSet]];
+
+MinimizedRuleSets[ruleSets_List] :=
+    RuleSetFromDNFF[BooleanMinimize[DNFFromRuleSet[ruleSets]]];
+
+RuleSetFromDNFF[minimizeBooleanExpression_] := Module[{RootExpressionToSubSet, ExpressionToSubSet, LeafExpressionToSubSet},
```

Comment:
You don't need this `Module` here. You can simply declare the functions outside of `RuleSetFromDNFF`.